src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -428,7 +430,14 @@ public SlackService newSlackService(AbstractBuild r, BuildListener listener) {
         authToken = env.expand(authToken);
         authTokenCredentialId = env.expand(authTokenCredentialId);
         room = env.expand(room);
-        return new StandardSlackService(baseUrl, teamDomain, authToken, authTokenCredentialId, botUser, room, false, r.getProject());
+        final String populatedToken = CredentialsObtainer.getTokenToUse(authTokenCredentialId, r.getProject(), authToken);
```

Comment:
`r.getParent()` is more correct but alas this is an `AbstractBuild` why is this named `r` 😭
Would you mind renaming the variable?